### ISSUE TITLE
Question about mini-batching / batch size

### ISSUE THREAD

**Issue by JeanKaddour** [rusty1s/pytorch_geometric#2112]
Hi, I'm new to GNNs, so sorry for any beginner mistakes in advance. 

I use mini-batching for both model training and testing. When I change the mini-batch size for the **TEST** dataloader, I observe varying test losses per example, all other things being equal. In particular, the higher the test mini-batch size, the higher the test loss per example. I don't understand why this is the case.

## MWE:

```
import math
import torch
import torch.nn.functional as F
from torch_geometric.data import Data, DataLoader
from torch_geometric.nn import GCNConv, global_mean_pool

# ---------------------------------------------------------------------------------------------------------------------
# -------------------------------------------------- Data generation --------------------------------------------------

edge_index = torch.tensor([[0, 1, 1, 2],
                           [1, 0, 2, 1]], dtype=torch.long)


def toy_node_features(some_number: float):
    feature = some_number / 3 / 2
    features = torch.tensor([[feature] * 2] * 3, dtype=torch.float)
    return features


N = 1000

data_list = []
for i in range(1, N):
    node_features = toy_node_features(i)
    data = Data(x=torch.tensor(node_features), edge_index=edge_index, y=torch.tensor(i, dtype=torch.float))
    data_list.append(data)


# ---------------------------------------------------------------------------------------------------------------------
# ------------------------------------------------------- Model -------------------------------------------------------


class ToyNet(torch.nn.Module):
    def __init__(self):
        super(ToyNet, self).__init__()
        self.conv1 = GCNConv(2, 16)
        self.conv2 = GCNConv(16, 1)

    def forward(self, data):
        x, edge_index, batch = data.x, data.edge_index, data.batch
        x = self.conv1(x, edge_index)
        x = F.relu(x)
        x = self.conv2(x, edge_index)
        x = global_mean_pool(x, batch=batch)

        return x


# ---------------------------------------------------------------------------------------------------------------------
# ---------------------------------------------------- Train / Test ---------------------------------------------------

def train(log_interval, model, device, train_loader, optimizer, epoch):
    model.train()

    for batch_idx, batch in enumerate(train_loader):
        batch = batch.to(device)
        optimizer.zero_grad()
        prediction = model(batch)
        outputs = batch.y
        loss = F.mse_loss(prediction, outputs)
        loss.backward()
        optimizer.step()
        # if batch_idx % log_interval == 0:
        #     print('Train Epoch: {} [{}/{} ({:.0f}%)]\tLoss: {:.6f}'.format(
        #         epoch, batch_idx * len(batch), len(train_loader.dataset),
        #                100. * batch_idx / len(train_loader), loss.item()))


def test(model, device, test_loader):
    model.eval()
    test_loss = 0.
    with torch.no_grad():
        for batch_idx, batch in enumerate(test_loader):
            batch = batch.to(device)
            outcome = batch.y
            pred_outcome = model(batch)
            test_loss += F.mse_loss(pred_outcome, outcome, reduction='sum').item()  # sum up batch loss
    test_loss /= len(test_loader.dataset)
    print(f'\nTest set: MSE: {test_loss:.5f}, RMSE: {math.sqrt(test_loss):.5f}')


# ---------------------------------------------------------------------------------------------------------------------
# -------------------------------------------------- Run --------------------------------------------------------------
TRAIN_BATCH_SIZE = 128
TEST_BATCH_SIZE = 32

device = torch.device("cpu")
torch.manual_seed(1)
train_loader = DataLoader(dataset=data_list[:800], batch_size=TRAIN_BATCH_SIZE)
test_loader = DataLoader(dataset=data_list[800:], batch_size=TEST_BATCH_SIZE)
model = ToyNet().to(device)
optimizer = torch.optim.Adam(model.parameters(), lr=0.01, weight_decay=5e-4)
test_interval = 20
log_interval = 20
for epoch in range(1, 100):
    train(log_interval, model, device, train_loader, optimizer, epoch)
    if epoch % test_interval == 0:
        test(model, device, test_loader)
```

**Comment by rusty1s** [comment]
There is a warning coming from PyTorch, denoting that prediction and ground-truth have different shapes, which leads to broadcasting issues. Fixable via `pred_outcome = pred_outcome.view(-1)`.

**Comment by JeanKaddour** [comment]
Arrgh. Thanks a million for your help, Matthias!
